eventsources/eventing.go

```diff
@@ -474,3 +557,16 @@ func generateClientID(hostname string) string {
 	clientID := fmt.Sprintf("client-%s-%v", strings.ReplaceAll(hostname, ".", "_"), r1.Intn(1000))
 	return clientID
 }
+
+func shouldDispatch(expression string, data map[string]interface{}) (bool, error) {
+	params := make(map[string]interface{})
+	for key, value := range data {
+		params[strings.ReplaceAll(key, "-", "_")] = value
+	}
+	env := expr.GetFuncMap(params)
+	return expr.EvalBool(strings.ReplaceAll(expression, "-", "_"), env)
+}
+
+func getFilterKey(eventType apicommon.EventSourceType, eventName string) string {
```

Comment:
`eventName` is unique in one EventSource object, so it's safe to use `eventName` directly as the map key, and remove this function.